### PR TITLE
Chime Cluster Definition and server code

### PR DESCRIPTION
**Pull request by andyg-apple**
Chime cluster definition and server code

### CODE REVIEW

**Comment by andyg-apple** [comment]
Will add generated code in second commit as soon as I get some feedback on this.

File: src/app/clusters/chime-server/chime-server.cpp
Comment (bzbarsky-apple) [inline src/app/clusters/chime-server/chime-server.cpp:1]: Ideally, the implementation would be a separate PR from adding the XML.

File: src/app/clusters/chime-server/chime-server.cpp [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+{
+    VerifyOrDie(aPath.mClusterId == Chime::Id);
+    
+    switch (aPath.mAttributeId)
+    {
+    case InstalledChimeSounds::Id:
+        return aEncoder.Encode(mDelegate.GetInstalledChimeSounds());
+
+    case ActiveChimeSoundId::Id:
+        return aEncoder.Encode(mDelegate.GetActiveChimeSoundId());
```
Comment (bzbarsky-apple) [inline src/app/clusters/chime-server/chime-server.cpp:79]: It would be better if the simple things that can be stored in the cluster were stored there, so we can mark things dirty properly on changes.

File: src/app/clusters/chime-server/chime-server.cpp [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+    bool activeSoundIdChanged = !(currentSoundId == soundId);
+
+    VerifyOrReturnError(activeSoundIdChanged, CHIP_NO_ERROR);
+    VerifyOrDie(mDelegate.SetActiveChimeSoundId(soundId) == CHIP_NO_ERROR);
+    MatterReportingAttributeChangeCallback(GetEndpointId(), Chime::Id, ActiveChimeSoundId::Id);
+
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR ChimeServer::SetEnabled(bool enabled)
```
Comment (bzbarsky-apple) [inline src/app/clusters/chime-server/chime-server.cpp:127]: OK, but the write path is not calling this?

File: src/app/clusters/chime-server/chime-server.h [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+ */
+class ChimeDelegate
+{
+public:
+    ChimeDelegate() = default;
+
+    virtual ~ChimeDelegate() = default;
+
+    // Get Attribute Methods
+    virtual DataModel::List<const Chime::Structs::ChimeSoundStruct::Type>&  GetInstalledChimeSounds() = 0;
```
Comment (bzbarsky-apple) [inline src/app/clusters/chime-server/chime-server.h:77]: What's the lifetime of the buffer backing that list?  How does the delegate know to not deallocate it?

Also, it it intentional that this implementation does not allow chunking the list?

File: src/app/zap-templates/zcl/data-model/chip/chime-cluster.xml [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+<configurator>
+  <domain name="CHIP"/>
+
+  <struct name="ChimeSoundStruct" apiMaturity="provisional">
+    <cluster code="0x0556"/>
+    <item name="ChimeId" type="int8u"/>
+    <item name="Name" type="char_string" max="48"/>
+  </struct>
+
+  <cluster apiMaturity="provisional">
```
Comment (bzbarsky-apple) [inline src/app/zap-templates/zcl/data-model/chip/chime-cluster.xml:26]: Where is the spec for this?  I don't see it in the Matter appclusters spec on "master"....

It's impossible to review this XML without the spec.

Comment (andyg-apple) [reply src/app/zap-templates/zcl/data-model/chip/chime-cluster.xml:26]: it's in the Cameras branch

File: src/app/clusters/chime-server/chime-server.cpp (hunk 2 above)
Comment (andyg-apple) [reply src/app/clusters/chime-server/chime-server.cpp:127]: Thanks, that's an error

**Comment by gmarcosb** [comment]
FYI chime cluster PR went in @ https://github.com/project-chip/connectedhomeip/pull/35806

Apologies did not realize you were working on this as well or would've reached out - started a chat on slack so we can coordinate
